### PR TITLE
Honor cc

### PR DESCRIPTION
**Pull request by jturney**
## Description
The ```setup``` script was not honoring the user's ```CC```, ```CXX```, or ```FC``` environment variables as it claimed to do.  Also, if compiler options were not set then allow CMake to decide.

## Todos
- [x] Honor environment variables.
- [x] Allow CMake to decide which compilers to use.

## Status
- [x]  Ready to go

### CODE REVIEW

**Comment by dgasmith** [comment]
Can you rebase out the merges first?

Can always 'git pull --rebase upstream master' to prevent them. 

-Daniel Smith

**Comment by dgasmith** [comment]
Can you rebase out the merges first? You can `git pull --rebase upstream master` to prevent them.

File: setup [hunk 1]
```diff
@@ -742,10 +742,15 @@ def configure_default_compilers(args):
             args.fc  and os.path.basename(args.fc).lower().startswith('openmpi'):
             args.mpi = 'on'
 
-    if not args.cc:
-        args.cc = 'gcc'
-    if not args.cxx:
-        args.cxx = 'g++'
+    if not args.cc and os.environ['CC']:
```
Comment (amjames) [inline setup]: These dictionary lookups will throw `KeyError` if there is no `--cxx/cc/fc` argument passed and the environmental variable is not set. 

It should probably catch that and either default to gcc/g++ (like before) or give some kind of message like:

> No C/C++/FORTRAN  compiler detected!   
 Set the environmental variable $CC/CXX/FC or use the --cc/cxx/fc flag at the command line

Comment (jturney) [inline setup]: Ah, none of my tests caught that. I'll get a fix committed very shortly. I had CC/CXX/FC defined.

**Pull request merged**